### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by aubinleyoung**
#### Used models to insert data

- Open rails console for your project.
- Created at least one user 
- Created at least 4 posts written by one of the users I created
- Created at least 6 posts comments for one of the posts I created 

#### Created custom methods

> Users

A method that returns the 3 most recent posts for a given user.

> Posts

A method that updates the posts counter for a user.
A method which returns the 5 most recent comments for a given post.

> Comments

A method that updates the comments counter for a post.

> Likes

A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class User < ApplicationRecord
+  private
+
+  def three_recent_posts
+    posts.order(created_at: :desc).limit(3)
+  end
+end
```
Comment (jerryowusu) [inline app/models/user.rb]: Kindly make sure that the associations between models are set up correctly according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png). 
For instance the user modes should have many posts, comments, and likes. 
Remember that author_id column in posts table should be the foreign_key for the users table. 
It should be something like this: 

```ruby
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Post < ApplicationRecord
+  def update_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
+
+  def five_recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+end
```
Comment (jerryowusu) [inline app/models/post.rb]: Kindly make sure that the associations between models are set up correctly according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png). 
For instance the post model should belong to author and has many comments and likes. It should be something like this:

```ruby
belongs_to :author, class_name: 'User'
has_many :comments
has_many :likes
```

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class Comment < ApplicationRecord
+  def update_comments_counter
+    post.update(comments_counter: post.comments.count)
+  end
+end
```
Comment (jerryowusu) [inline app/models/comment.rb]: Kindly make sure that the associations between models are set up correctly according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png). 
For instance, the comment model should belong to author and post. 
It should be something like this: 

```ruby
belongs_to :author, class_name: 'User'
belongs_to :post
```

File: app/models/like.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class Like < ApplicationRecord
+  def update_likes_counter
+    post.update(likes_counter: post.likes.count)
+  end
+end
```
Comment (jerryowusu) [inline app/models/like.rb]: Kindly make sure that the associations between models are set up correctly according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png). 
For instance, the like model should belong to author and post.
Something like this: 

```ruby
belongs_to :author, class_name: 'User'
belongs_to :post
```

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi @aubinleyoung 

Good job so far!
Your project is looking fantastic so far :star_struck: 

However, there are few changes you need to make in order to move on to the next level :sunglasses:

almost there


## Highlights
Nice PR description :+1: 
Nice commit messages :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Qoosim — APPROVED** [review]
#  *Status: Approved* :tada: :heavy_check_mark: 

Hi @aubinleyoung, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**